modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/agg/Accumulators.java

```diff
@@ -88,11 +88,11 @@ private Supplier<Accumulator> accumulatorFunctionFactory(AggregateCall call) {
             case "SOME":
                 return minMaxFactory(false, call);
             case "SINGLE_VALUE":
-                return SingleVal.FACTORY;
-            case "LITERAL_AGG":
-                return LiteralVal.FACTORY;
+                return singleAnyValueFactory(true, call);
             case "ANY_VALUE":
-                return AnyVal.FACTORY;
+                return singleAnyValueFactory(false, call);
```

Comment:
i think new division by singleAnyValueFactory(true|false become less understandable ?

Reply:
I divided it into two factories, if I understood your comment correctly